arm-sql/2014-04-01/swagger/serverFirewallRules.json

```diff
@@ -0,0 +1,268 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Server Firewall Rule APIs",
+    "description": "The Azure SQL Database management API provides a RESTful set of web services that interact with Azure SQL Database services to manage your server firewall rules. The API enables users to create, retrieve, update, and delete server firewall rules.",
+    "version": "2014-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/firewallRules/{firewallRule}": {
+      "put": {
+        "tags": [
+          "FirewallRules"
+        ],
+        "operationId": "Servers_CreateOrUpdateFirewallRule",
```

Comment:
This was originally a deliberate decision to nest firewall rules methods inside the SqlServer class in generated code to reduce the number of top-level classes and provide better structure. The thinking was that a firewall rule is a very simple nested resource so it doesn’t need to be a full-fledged top level class in generated code. Are you recommending we change this plan? How do other RPs manage having a large number of resource types?